R/redist_ms_parallel.R

```diff
@@ -161,22 +161,45 @@ redist_mergesplit_parallel = function(map, nsims, chains=1, warmup=floor(nsims/2
     on.exit(stopCluster(cl))
 
     each_len = if (return_all) nsims - warmup else 1
-    plans = foreach(chain=seq_len(chains), .combine=cbind) %dopar% {
+    out_par <- foreach(chain=seq_len(chains)) %dopar% {
         if (!silent) cat("Starting chain ", chain, "\n", sep="")
-        algout = ms_plans(nsims+1L, adj, init_plans[, chain], counties, pop,
-                          ndists, pop_bounds[2], pop_bounds[1], pop_bounds[3],
-                          compactness, constraints, adapt_k_thresh, k, verbosity)
-        if (return_all)
+        ms_plans(nsims+1L, adj, init_plans[, chain], counties, pop,
+                 ndists, pop_bounds[2], pop_bounds[1], pop_bounds[3],
+                 compactness, constraints, adapt_k_thresh, k, verbosity)
+    }
+
+    plans <- lapply(out_par, function(algout) {
+        if (return_all) {
             algout$plans[, -1:-(warmup+1L), drop=FALSE]
-        else
+        } else {
             algout$plans[, nsims+1L, drop=FALSE]
-    }
-    out = new_redist_plans(plans, map, "mergesplit", NULL, FALSE,
+        }
+    })
+    plans <- do.call('cbind', plans)
+
+    mh <- sapply(out_par, function(algout) {
+        mean(as.logical(algout$mhdecisions))
+    })
+
+    acceptances <- sapply(out_par, function(algout) {
+        if (!return_all) {
+            as.logical(algout$mhdecisions[nsims])
+        } else if (warmup == 0) {
+            as.logical(algout$mhdecisions)
+        } else {
+            as.logical(algout$mhdecisions[-seq_len(warmup)])
```

Comment:
why shouldn't this match the indexes above (ll. 173 and 175)? seems like it's off by 1?

Reply:
`mh_decisions` is one entry shorter than `ncol(plans)`

Reply:
gotcha